backend/dataall/modules/shares_base/db/share_object_repositories.py

```diff
@@ -10,11 +10,9 @@
 from dataall.modules.datasets_base.db.dataset_models import DatasetBase
 from dataall.modules.datasets_base.db.dataset_repositories import DatasetBaseRepository
 from dataall.modules.notifications.db.notification_models import Notification
+from dataall.modules.s3_datasets.db.dataset_models import S3Dataset
```

Comment:
Can you also remove the import to S3Dataset? It is not used and it imports parts of s3_datasets, with the potential risk of circular dependencies